### PR TITLE
[6463] Fixed autocomplete undefined issue

### PR DESCRIPTION
**Pull request by defong**
### Context
Autocomplete undefined

### Changes proposed in this pull request
Fixed autocomplete undefined issue

#### Before
image

#### After
image

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by defong** [comment]
[Bug party 1: Undefined value in autocomplete when editing existing placement](https://trello.com/c/Uh5GgDLr/6463-bug-party-1-undefined-value-in-autocomplete-when-editing-existing-placement)

**Comment by andrewscrivener** [comment]
Working well @defong :shipit:

**Review by carlosmartinez — APPROVED** [review]

File: app/javascript/scripts/components/form_components/providers_autocomplete.js [hunk 1]
```diff
@@ -83,14 +83,16 @@ const setProviderHiddenField = (value) => {
...
 
-const suggestionTemplate = (value) => {
-  if (!value) {
-    return
-  }
-
-  const hints = [`Code ${value.code}`, value.name, value.ukprn].filter(Boolean)
-
-  return `${value.name} <span class="autocomplete__option--hint">${hints.join(', ')}</span>`
+const suggestionTemplate = (result) => {
```
Comment (darokel) [inline app/javascript/scripts/components/form_components/providers_autocomplete.js:86]:
```suggestion
const suggestionTemplate = (result) => {
  if (!result) return "";

  if (typeof result === "string") return result;

  const hints = [`Code ${result.code}`, result.name, result.ukprn].filter(Boolean);

  return `${result.name} <span class="autocomplete__option--hint">${hints.join(", ")}</span>`;
};
```

Comment (defong) [reply app/javascript/scripts/components/form_components/providers_autocomplete.js:86]: Prefer not doing return statement in that manner as we be at the mercy of the complex object if it clears the first few hurdles.

**Pull request merged**